lib/puppet_x/puppetlabs/powershell_manager.rb

```diff
@@ -38,11 +38,26 @@ def self.supported?
         !win32console_enabled?
       end
 
+      def check_valid_paths(paths)
```

Comment:
If all we are doing is returning a boolean, we could conceivably change this to something like `def invalid_lib_paths?` and the call below could shorten to `raise <error message> if invalid_lib_paths?`

I also think that since `ENV['lib']`, as part of the environment, is necessarily global to all of the code in this file, there's no real need to pass it as a parameter. You can access it directly from within this function if the name is changed as above to make it clear that you are testing the `lib` variable specifically.